assets/src/edit-story/components/library/panes/text/textSets/textSets.js

```diff
@@ -32,34 +32,66 @@ import { __ } from '@wordpress/i18n';
 import { Section } from '../../../common';
 import { UnitsProvider } from '../../../../../units';
 import { PAGE_RATIO, TEXT_SET_SIZE } from '../../../../../constants';
+import PillGroup from '../../shared/pillGroup';
+import { useRovingTabIndex } from '../../shared';
 import { getTextSets } from './utils';
 import TextSet from './textSet';
 
 const TextSetContainer = styled.div`
   display: grid;
   grid-template-columns: 1fr 1fr;
   row-gap: 12px;
+  column-gap: 12px;
+  margin-top: 28px;
+`;
+
+/* Undo the -1.5em set by the Pane */
+const CategoryWrapper = styled.div`
+  margin-left: -1.5em;
+  margin-right: -1.5em;
 `;
 
 function TextSets() {
   const [textSets, setTextSets] = useState([]);
+  const [selectedCat, setSelectedCat] = useState(null);
+  const ref = useRef();
+
+  const getFilteredTextSets = useCallback(() => {
+    if (selectedCat && textSets[selectedCat]) {
+      return textSets[selectedCat];
+    }
+    return Object.values(textSets).flat();
+  }, [selectedCat, textSets]);
 
   useEffect(() => {
     getTextSets().then(setTextSets);
   }, []);
 
+  useRovingTabIndex({ ref });
```

Comment:
This just worked? That's pretty freaking awesome! 🎉  🎉

Reply:
Indeed!